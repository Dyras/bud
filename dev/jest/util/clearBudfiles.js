const {noop} = require('lodash')
const rimraf = require('rimraf')

module.exports = async () => {
  return rimraf(
    `${process.cwd()}/examples/*/.budfiles`,
    {
      glob: true,
    },
    noop,
  )
}
