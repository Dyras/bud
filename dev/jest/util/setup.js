/* eslint-disable no-console */
// @ts-check

const chalk = require('chalk')
const execa = require('execa')

const clearArtifacts = require('./clearArtifacts')
const paths = require('./paths')

const task = async (cmd, paths) => {
  console.log(chalk.blue`\nyarn bud ${cmd}\n`)

  await Promise.all(
    paths.map(async ex => {
      const task = execa.command(`yarn bud ${cmd}`, {
        cwd: ex.cwd,
      })
      task.stdout.pipe(process.stdout)
      await task
    }),
  )
}

module.exports = async () => {
  await clearArtifacts()

  const examples = await paths()

  await task('build --no-dashboard', examples)
}
