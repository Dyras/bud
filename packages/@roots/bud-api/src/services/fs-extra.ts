import {fs} from '@roots/bud-support'

export const ensureDirSync = fs.ensureDirSync
export const pathExistsSync = fs.pathExistsSync
