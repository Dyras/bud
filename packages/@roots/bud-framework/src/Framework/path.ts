import {join} from 'path'

import {Framework} from '..'

interface path {
  (
    this: Framework,
    key: keyof Framework.Locations,
    ...path: string[]
  ): string
}

function path(
  key: keyof Framework.Locations,
  ...path: string[]
): string {
  return join(
    ...[
      key !== 'project'
        ? this.hooks.filter('location/project')
        : false,
      this.hooks.filter(`location/${key}`),
      ...(path ?? []),
    ].filter(Boolean),
  )
}

export {path}
