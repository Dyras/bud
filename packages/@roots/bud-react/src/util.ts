/**
 * @module @roots/bud-react
 */

import type {Configuration} from 'webpack'

/**
 * @function addRefresh
 */
function addRefresh(entries, [name, assets]) {
  return {
    ...(entries ?? {}),
    [name]: {
      ...assets,
      import: [
        'react-refresh/runtime',
        ...(assets.import ?? []),
      ],
    },
  }
}

/**
 * @function reducer
 */
function reducer(entry: Configuration['entry']) {
  return entry
    ? Object.entries(entry).reduce(addRefresh, entry)
    : {}
}

/**
 * @exports default
 */
export default reducer
