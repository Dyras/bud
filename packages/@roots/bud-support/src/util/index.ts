export {notify} from './notify'
export {dump} from './dump'
export {format} from './format'
export {processHandler} from './processHandler'

import InterpolateHtmlPlugin from './InterpolateHtmlPlugin'
export {InterpolateHtmlPlugin}
