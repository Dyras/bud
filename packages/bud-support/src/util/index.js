"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.InterpolateHtmlPlugin = exports.formatWebpackMessages = exports.eslintFormatter = exports.processHandler = exports.format = exports.dump = exports.notify = void 0;
var notify_1 = require("./notify");
Object.defineProperty(exports, "notify", { enumerable: true, get: function () { return notify_1.notify; } });
var dump_1 = require("./dump");
Object.defineProperty(exports, "dump", { enumerable: true, get: function () { return dump_1.dump; } });
var format_1 = require("./format");
Object.defineProperty(exports, "format", { enumerable: true, get: function () { return format_1.format; } });
var processHandler_1 = require("./processHandler");
Object.defineProperty(exports, "processHandler", { enumerable: true, get: function () { return processHandler_1.processHandler; } });
const eslintFormatter_1 = __importDefault(require("./eslintFormatter"));
exports.eslintFormatter = eslintFormatter_1.default;
const InterpolateHtmlPlugin_1 = __importDefault(require("./InterpolateHtmlPlugin"));
exports.InterpolateHtmlPlugin = InterpolateHtmlPlugin_1.default;
const formatWebpackMessages_1 = __importDefault(require("./formatWebpackMessages"));
exports.formatWebpackMessages = formatWebpackMessages_1.default;
//# sourceMappingURL=index.js.map