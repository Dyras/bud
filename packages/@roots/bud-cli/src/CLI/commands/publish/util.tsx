import {chalk, globby} from '@roots/bud-support'
import {dirname} from 'path'

/**
 * Publishable matches (glob pattern)
 */
export const publishables = [
  ...globby
    .sync([
      '**/node_modules/bud-*/publish/*',
      '**/node_modules/*/bud-*/publish/*',
    ])
    .filter(
      (value, index, self) => self.indexOf(value) === index,
    ),
]

export const formatted = publishables.reduce((a, file) => {
  const ext = dirname(dirname(file)).replace('node_modules/', '')

  const template = file
    .replace(`${ext}/publish/`, '')
    .replace('node_modules/', '')

  return `${a} \n${chalk.blue(ext)} ${chalk.green(template)}`
}, ``)
