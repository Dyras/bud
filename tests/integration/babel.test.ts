import {readFile, readJson} from 'fs-extra'
import {join} from 'path'

import {Assets, helper} from '../util/integration'

const suite = helper('babel', 'examples/babel')

jest.setTimeout(60000)

describe(suite.name, () => {
  let assets: Assets

  beforeAll(async () => {
    assets = await suite.setup()
  })

  describe('app.js', () => {
    it('has contents', () => {
      expect(assets['app.js'].length).toBeGreaterThan(10)
    })

    it('is transpiled', () => {
      expect(assets['app.js'].includes('import')).toBeFalsy()
    })
  })

  describe('snapshots', () => {
    it('dist/manifest.json', async () => {
      const artifact = await readFile(
        join(process.cwd(), 'examples/babel/dist/manifest.json'),
      )

      expect(artifact.toString()).toMatchSnapshot()
    })

    it('dist/app.css', async () => {
      const artifact = await readFile(
        join(process.cwd(), 'examples/babel/dist/app.css'),
      )

      expect(artifact.toString()).toMatchSnapshot()
    })

    it('dist/app.js', async () => {
      const artifact = await readFile(
        join(process.cwd(), 'examples/babel/dist/app.js'),
      )

      expect(artifact.toString()).toMatchSnapshot()
    })

    it('.budfiles/bud.webpack.config.js', async () => {
      let artifact = await import(
        join(
          process.cwd(),
          'examples/babel/.budfiles/bud.webpack.config.js',
        )
      )

      artifact = artifact()

      expect(artifact.entry).toMatchSnapshot()
      expect(artifact.mode).toMatchSnapshot()
      expect(artifact.optimization).toMatchSnapshot()
      expect(artifact.bail).toMatchSnapshot()
      expect(artifact.cache).toMatchSnapshot()
    })
  })

  it('module map matches snapshot', async () => {
    const artifact = await readJson(
      join(
        process.cwd(),
        'examples/babel/.budfiles/bud-modules.json',
      ),
    )

    expect(artifact.chunks).toMatchSnapshot({
      byName: {
        app: expect.any(Number),
      },
      bySource: {
        '0 app': expect.any(Number),
      },
      usedIds: [expect.any(Number)],
    })
  })
})
