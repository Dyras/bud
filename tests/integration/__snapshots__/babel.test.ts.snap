// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples/babel .budfiles/bud.webpack.config.js 1`] = `
Object {
  "app": Object {
    "import": Array [
      "app.js",
      "demo.js",
      "foo.css",
      "global.css",
    ],
  },
}
`;

exports[`examples/babel .budfiles/bud.webpack.config.js 2`] = `"production"`;

exports[`examples/babel .budfiles/bud.webpack.config.js 3`] = `
Object {
  "minimize": true,
  "minimizer": Array [
    "...",
    Object {
      "options": Object {
        "minify": "<<Function 'cssnanoMinify'>>",
        "minimizerOptions": Object {
          "preset": Array [
            "default",
            Object {
              "discardComments": Object {
                "removeAll": true,
              },
            },
          ],
        },
        "parallel": true,
        "test": /\\\\\\.css\\(\\\\\\?\\.\\*\\)\\?\\$/i,
        "warningsFilter": "<<Function 'warningsFilter'>>",
      },
    },
    Object {
      "options": Object {
        "minify": "<<Function 'cssnanoMinify'>>",
        "minimizerOptions": "<<Circular reference to 'config.optimization.minimizer.[1].options.minimizerOptions'>>",
        "parallel": true,
        "test": /\\\\\\.css\\(\\\\\\?\\.\\*\\)\\?\\$/i,
        "warningsFilter": "<<Function 'warningsFilter'>>",
      },
    },
  ],
  "moduleIds": "deterministic",
  "splitChunks": Object {
    "cacheGroups": Object {
      "vendor": Object {
        "chunks": "all",
        "filename": "vendor/[name].js",
        "priority": -10,
        "reuseExistingChunk": true,
        "test": /\\[\\\\\\\\/\\]node_modules\\[\\\\\\\\/\\]/,
      },
    },
    "defaultSizeTypes": Array [
      "...",
    ],
  },
}
`;

exports[`examples/babel .budfiles/bud.webpack.config.js 4`] = `true`;

exports[`examples/babel .budfiles/bud.webpack.config.js 5`] = `
Object {
  "type": "memory",
}
`;

exports[`examples/babel app.css matches snapshot 1`] = `"p{font-weight:700}body,html{margin:0;padding:0}.app{align-items:center;background:#5813d5;color:#fff;display:flex;font-family:sans-serif;height:100vh;justify-content:center;letter-spacing:.2em;text-align:center;text-transform:uppercase;width:100vw}"`;

exports[`examples/babel app.js matches snapshot 1`] = `"!function(){var n={344:function(n,e,o){\\"use strict\\";o(755),document.querySelector(\\"body\\").innerHTML=\\"\\\\n  <div>\\\\n    <h1>Hello from babel!</h1>\\\\n  </div>\\\\n\\"},755:function(){[\\"fancy\\",\\"ecma\\"].map((function(n){console.log(n)}))},615:function(){},968:function(){}},e={};function o(t){var r=e[t];if(void 0!==r)return r.exports;var u=e[t]={exports:{}};return n[t](u,u.exports,o),u.exports}o.n=function(n){var e=n&&n.__esModule?function(){return n.default}:function(){return n};return o.d(e,{a:e}),e},o.d=function(n,e){for(var t in e)o.o(e,t)&&!o.o(n,t)&&Object.defineProperty(n,t,{enumerable:!0,get:e[t]})},o.o=function(n,e){return Object.prototype.hasOwnProperty.call(n,e)},o(344),o(755),o(615),o(968)}();"`;

exports[`examples/babel manifest.json 1`] = `
Object {
  "app.css": "app.css",
  "app.js": "app.js",
}
`;

exports[`examples/babel module map matches snapshot 1`] = `
Object {
  "byName": Object {
    "app": Any<Number>,
  },
  "bySource": Object {
    "0 app": Any<Number>,
  },
  "usedIds": Array [
    Any<Number>,
  ],
}
`;
